runtime/memory/64bit_nix.c

```diff
@@ -37,6 +37,19 @@ void expand_memory() {
     memory_size += WASM_PAGE_SIZE;
 }
 
+i32 instruction_memory_size() {
+    return memory_size / WASM_PAGE_SIZE;
```

Comment:
Assert that `memory_size` is a multiple of `WASM_PAGE_SIZE` so that we can make sure we don't have round down errors here?

Reply:
Offhand comment: Seems strange that this instruction returned a signed value.

Reply:
I don't think signed vs unsigned matters for correctness (WASM has no type level distinction), maybe u32 better for clearness?

Reply:
Hmm, I think it's for consistency with memory.grow, which can return negative error codes. Since memory.grow uses i32, memory.size can't use the range >2^31 anyways.

Would we ever expect memory.size to return an error? The Wasm spec suggests no. I can change it next batch of changes if you all think we should.

Spec here:
https://webassembly.github.io/spec/core/exec/instructions.html#xref-syntax-instructions-syntax-instr-memory-mathsf-memory-size

Reply:
Just noticed @gparmer's original comment. Can add assert 👍

Reply:
Second pass: I think rounding down is actually the correct behavior. Nothing in the memory instructions strictly requires the page alignment. The only mechanisms available right now are limited to page boundaries, but if you provided something like a custom sbrk as a separate import (is this what wasmception does?), you could expand your memory by a non-page amount. This would work as long as the memory backend supports it.

If, say, you had 1.5 pages allocated, it seems accurate for `memory.size` to return 1. The code would then know you have at least 1 page. It might cause problems for users experimenting with non-page memory sizes if the `memory.size` instruction asserted.

Reply:
@geky I don't necessarily agree, since the spec says that `memory` is always divisible by page size. It seems our conclusion is to punt on that